core/src/avm2/globals/flash/display/loaderinfo.rs

```diff
@@ -242,14 +242,18 @@ pub fn url<'gc>(
 ) -> Result<Value<'gc>, Error> {
     if let Some(this) = this {
         if let Some(loader_stream) = this.as_loader_stream() {
-            match &*loader_stream {
-                LoaderStream::Stage => {
-                    return Err("Error: The stage's loader info does not have a URL".into())
-                }
-                LoaderStream::Swf(root, _) => {
-                    let url = root.url().unwrap_or("");
-                    return Ok(AvmString::new_utf8(activation.context.gc_context, url).into());
-                }
+            let root = match &*loader_stream {
+                LoaderStream::Stage => activation
+                    .context
+                    .stage
+                    .child_by_index(0)
+                    .and_then(|c| c.movie()),
```

Comment:
This isn't quite right because you can do:
```
var s:Stage = stage;
trace(s.loaderInfo);
trace(s.loaderInfo.url);

s.removeChildAt(0);

// Still valid
trace(s.loaderInfo);
trace(s.loaderInfo.url);
```
(also this is a more general problem in that panic if there isn't a root clip on the stage).